File: ruby/10.rb
# project euler 10: find the sum of all primes below 2,000,000
require 'prime'
t = Time.new.to_f
sum = Prime.take_while {|p| p < 2000000}.inject {|sum,p| sum + p}
puts "result: #{sum} - found in #{((Time.new.to_f - t)*1000).round 3} ms"

File: lua/10.lua
--project euler <n>: <description>
package.path = "..\\libraries\\?.lua;" .. package.path
require "lib"
local t = os.clock()

local n = 1
local sum = 1

repeat
	if (lib.is_prime(n)) then
		sum = sum+n
	end
	
	if (n <= 2) then
		n = n+1
	else
		n = n+2
	end
until (n == 2e6-1)

print("result: "..sum.." - found in "..(os.clock() - t).." s")